### PR TITLE
feat(roles): refactor get_top_dogs

### PR DESCRIPTION
**Pull request by cathteng**
Get all of the top dogs including through team membership

Refactor for `ProjectTransfer`, `get_owners()` in the `Organization` model, and `get_emails_for_user_or_org` in `commits`

For ER-1365

### CODE REVIEW

File: src/sentry/api/endpoints/project_transfer.py [hunk 1]
```diff
@@ -59,10 +60,21 @@ def post(self, request: Request, project) -> Response:
...
-                {"detail": "Could not find an organization owner with that email"},
-                status=status.HTTP_404_NOT_FOUND,
-            )
+            try:
+                org_members = list(
+                    OrganizationMember.objects.filter(
+                        user__email__iexact=email, user__is_active=True
+                    ).values_list("id", flat=True)
+                )
+                org_member_id = OrganizationMemberTeam.objects.filter(
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/project_transfer.py:69]: how does this work exactly? so someone could be an org level "member", but then be on a team that is owner? i searched briefly for docs in the code and didn't see anything, mind pointing me somewhere?

Comment (cathyteng17) [reply src/sentry/api/endpoints/project_transfer.py:69]: yes, that's right. you can be on a team that has an org role that extends the scopes of your user-set organization role https://github.com/getsentry/sentry/blob/master/src/sentry/models/organizationmember.py#L398

File: src/sentry/models/organization.py [hunk 2]
```diff
@@ -283,11 +283,22 @@ def get_audit_log_data(self):
...
+            OrganizationMemberTeam.objects.filter(
+                team_id__in=owner_teams,
+            ).values_list("organizationmember_id", flat=True)
+        )
 
-        owner_memberships = OrganizationMember.objects.filter(
-            role=roles.get_top_dog().id, organization=self
+        # get owners from org role
+        owners = self.member_set.filter(
+            Q(role=roles.get_top_dog().id) | Q(id__in=owner_team_members)
```
Comment (JoshFerge) [inline src/sentry/models/organization.py:298]: it looks like the `organization=self` filter on this query was removed. is this intentional?

Comment (cathyteng17) [reply src/sentry/models/organization.py:298]: yes, `self.member_set` is the set of all members of the organization. it's also used here https://github.com/getsentry/sentry/blob/master/src/sentry/models/organization.py#L269

File: src/sentry/tasks/commits.py [hunk 3]
```diff
@@ -246,10 +246,26 @@ def is_integration_provider(provider):
...
+        # get owners from owner teams
+        owner_team_members = list(
+            OrganizationMemberTeam.objects.filter(
+                team_id__in=owner_teams,
+            ).values_list("organizationmember_id", flat=True)
+        )
+
         members = OrganizationMember.objects.filter(
-            organization_id=orgId, role="owner", user_id__isnull=False
+            Q(role="owner") | Q(id__in=owner_team_members),
```
Comment (JoshFerge) [inline src/sentry/tasks/commits.py:266]: i'm curious why we use `top_tog` some places, and `owner` directly others.

File: src/sentry/models/organization.py (hunk 2 above)
Comment (JoshFerge) [reply src/sentry/models/organization.py:298]: ah got it 👍🏼

File: src/sentry/models/organization.py [hunk 4]
```diff
@@ -131,37 +131,29 @@ def get_organizations_where_user_is_owner(
...
-
         # get owners from orgs
         owner_role_orgs = set(
-            OrganizationMember.objects.filter(
-                role=roles.get_top_dog().id, user_id=user_id
-            ).values_list("organization_id", flat=True)
+            members.filter(role=roles.get_top_dog().id).values_list("organization_id", flat=True)
         )
 
         return self.filter(id__in=orgs.union(owner_role_orgs), status=OrganizationStatus.ACTIVE)
```
Comment (cathyteng17) [inline src/sentry/models/organization.py:159]: the above refactor simplifies this whole function

**Review by RyanSkonnord — APPROVED** [review]
Looks good; I'm glad to see this simplified and streamlined. Just a few nits.

**Pull request merged**